### PR TITLE
Implemented functionality to display PO ids and filter order on the basis of PO ids(#2335pf5)

### PR DESCRIPTION
**Pull request by ymaheshwari1**

### CODE REVIEW

File: src/store/modules/order/actions.ts [hunk 1]
```diff
@@ -218,6 +222,11 @@ const actions: ActionTree<OrderState, RootState> = {
       payload.json.filter = payload.json.filter.concat(` AND autoCancelDate: [${state.currentOrderFiltersSelected.autoCancelDate + 'T00:00:00Z'} TO ${state.currentOrderFiltersSelected.autoCancelDate + 'T23:59:59Z'}]`)
     }
 
+    const correspondingPoId = state.currentOrderFiltersSelected.poIds.map((id: string) => state.availableOrderFilterOptions.poIds[id]).toString().replaceAll(",", " OR ")
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts]: We could use join() method with ' OR ' as separator
https://stackoverflow.com/questions/39771131/how-to-convert-array-into-comma-separated-string-in-javascript

File: src/store/modules/order/actions.ts [hunk 2]
```diff
@@ -122,6 +121,11 @@ const actions: ActionTree<OrderState, RootState> = {
     return payload;
   },
 
+  async availableFilterOptionsUpdated({ commit }, payload) {
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts]:
```suggestion
  async updateAvailableFilterOptions({ commit }, payload) {
```

File: src/components/OrderFilters.vue [hunk 3]
```diff
@@ -130,6 +130,11 @@ export default defineComponent({
   },
   methods: {
     async appliedFiltersUpdated(value: string, filterName: string) {
+      const poIds = this.currentOrderFiltersSelected.poIds;
+      if (filterName === 'poIds') {
```
Comment (adityasharma7) [inline src/components/OrderFilters.vue]: I think this should be part of actions, as if are updating the poIds from multiple instance we will have to handle at multiple places, also we may have more such filters

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/order/actions.ts (hunk 2 above)
Comment (ymaheshwari1) [reply src/store/modules/order/actions.ts]: Sir, Updated the action name.

File: src/components/OrderFilters.vue (hunk 3 above)
Comment (ymaheshwari1) [reply src/components/OrderFilters.vue]: Done Sir

File: src/store/modules/order/actions.ts (hunk 1 above)
Comment (ymaheshwari1) [reply src/store/modules/order/actions.ts]: Replaced the usage of replaceAll with join

File: src/store/modules/order/actions.ts [hunk 4]
```diff
@@ -117,11 +116,21 @@ const actions: ActionTree<OrderState, RootState> = {
     return resp;
   },
 
-  async appliedFiltersUpdated({ commit }, payload) {
+  async appliedFiltersUpdated({ state, commit }, payload) {
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts]:
```suggestion
  async updateAppliedFilters({ state, commit }, payload) {
```

File: src/views/FindOrder.vue [hunk 5]
```diff
@@ -259,11 +259,31 @@ export default defineComponent ({
...
+        "fields": "externalOrderId orderId",
+        "query": "*:*"
       }
-    })
+    }
+
+    const resp = await OrderService.getPOIds(payload);
+    if (resp.status == 200 && !hasError(resp)) {
+      resp.data.grouped.externalOrderId.groups.map((group: any) => {
+        if (group.groupValue) {
```
Comment (adityasharma7) [inline src/views/FindOrder.vue]: We could try with solr query that we exclude empty values

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/FindOrder.vue (hunk 5 above)
Comment (ymaheshwari1) [reply src/views/FindOrder.vue]: Sir, filtered results using solr query.

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**